astropy/cosmology/core.py

```diff
@@ -67,18 +67,16 @@
 #  and reuse in the initialization rather than have every object do them
 # Note that the call to cgs is actually extremely expensive,
 #  so we actually skip using the units package directly, and
-#  hardwire the conversion from mks to cgs. This assumes that constants
-#  will always return mks by default -- if this is made faster for simple
+#  hardwire the conversion from mks to cgs. If this is made faster for simple
 #  cases like this, it should be changed back.
-# Note that the unit tests should catch it if this happens
 H0units_to_invs = (u.km / (u.s * u.Mpc)).to(1.0 / u.s)
 sec_to_Gyr = u.s.to(u.Gyr)
 # const in critical density in cgs units (g cm^-3)
-critdens_const = 3. / (8. * pi * const.G.value * 1000)
+critdens_const = 3. / (8. * pi * const.G.to_value(u.m**3 / u.kg / u.s**2) * 1000)
```

Comment:
Since this part is not performance critical, how about just doing `const.G.si.value`? Or, better still, since `cgs` is wanted, 
`(3/(8*pi*const.G)).cgs.value`?